src/core/main.c

```diff
@@ -924,26 +924,32 @@ static int parse_argv(int argc, char *argv[]) {
         if (getpid_cached() == 1)
                 opterr = 0;
 
-        while ((c = getopt_long(argc, argv, SYSTEMD_GETOPT_SHORT_OPTIONS, options, NULL)) >= 0)
+        /* Reset optind to restart scanning in case getopt_long() already parsed argv. */
+        if (on_reload)
+                optind = 1;
```

Comment:
We usually reset optind to 0, to make sure all glibc special prefixes are honored too

Reply:
The man page explicitly documents "1" as the value to use to restart options scanning regardless of whether any GNU extension are enabled or not (if that's what you meant).